src/runner.cpp

```diff
@@ -0,0 +1,32 @@
+#include <cloysterhpc/functions.h>
+#include <cloysterhpc/runner.h>
+#include <cloysterhpc/services/log.h>
+#include <fmt/format.h>
+
+int BaseRunner::downloadFile(const std::string& url, const std::string& file)
+{
+    auto cmd = fmt::format("wget -NP {} {}", file, url);
+    return this->executeCommand(cmd);
+}
+
+int Runner::executeCommand(const std::string& cmd)
+{
+    return cloyster::runCommand(cmd);
+}
+
+int DryRunner::executeCommand(const std::string& cmd)
+{
+    LOG_INFO("Would execute command: {}", cmd);
+    return 0;
+}
+
+int MockRunner::executeCommand(const std::string& cmd)
+{
+    m_cmds.push_back(cmd);
+    return 0;
```

Comment:
Since this is the only return does it makes sense? It always `return 0`.

Reply:
Yes

This class descends from the `BaseRunner`. One of its child is the normal `Runner`, which really runs a command and returns a value depending on said command return code